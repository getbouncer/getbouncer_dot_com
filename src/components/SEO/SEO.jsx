import React from 'react'
import { Helmet } from 'react-helmet'
import PropTypes from 'prop-types'
import { useStaticQuery, graphql } from 'gatsby'

const SEO = ({ desc, pathname, title }) => {
  const { site } = useStaticQuery(graphql`
    query SEO {
      site {
        siteMetadata {
          siteUrl
          defaultTitle: title
          defaultDescription: description
        }
      }
    }
  `)

  const {
    siteMetadata: { defaultDescription, defaultTitle, siteUrl },
  } = site

  const seo = {
    title: title || defaultTitle,
    description: desc || defaultDescription,
    url: `${siteUrl}${pathname || ''}`,
  }

  return (
    <>
      <Helmet title={seo.title}>
        <html lang="en" />
        <meta name="description" content={seo.description} />
        <meta name="bouncer" content="Bouncer Marketing Website" />
      </Helmet>
    </>
  )
}

export default SEO

SEO.propTypes = {
  title: PropTypes.string,
  desc: PropTypes.string,
  pathname: PropTypes.string,
}

SEO.defaultProps = {
  title: null,
  desc: null,
  pathname: null,
}
