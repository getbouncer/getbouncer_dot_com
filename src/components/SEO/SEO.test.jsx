import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, it } from 'jest'
import { PureSEO as SEO } from './SEO'

describe('SEOComponent', () => {
  it('renders correctly', () => {
    const site = {
      siteMetadata: {
        siteUrl: 'https://www.example.com/',
        defaultTitle: 'defaultTitle',
        defaultDescription: 'defaultDescription',
      },
    }
    const tree = renderer
      .create(<SEO site={site} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
