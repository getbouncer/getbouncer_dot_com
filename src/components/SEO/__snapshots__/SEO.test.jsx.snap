// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEOComponent renders correctly 1`] = `null`;
